src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -33,10 +33,14 @@ public DeleteCommand parse(String args) throws ParseException {
                     && !arePrefixesPresent(argMultimap, PREFIX_EMAIL);
             boolean isBothFilled = arePrefixesPresent(argMultimap, PREFIX_PHONE)
                     && arePrefixesPresent(argMultimap, PREFIX_EMAIL);
+            boolean isPrefixUnique = arePrefixesPresent(argMultimap, PREFIX_PHONE)
+                    ? argMultimap.isUniquePrefix(PREFIX_PHONE)
+                    : argMultimap.isUniquePrefix(PREFIX_EMAIL);
 
             if (isBothEmpty
                     || isBothFilled
-                    || !argMultimap.getPreamble().isEmpty()) {
+                    || !argMultimap.getPreamble().isEmpty()
+                    || !isPrefixUnique) {
                 throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE));
```

Comment:
I like how this manages to catch user errors in inputting unwanted duplicate prefixes, but would it be possible to abstract this out into the [`getValue()` method](https://github.com/AY2223S1-CS2103T-W09-1/tp/pull/84/files#:~:text=public%20Optional%3CString%3E%20getValue(Prefix%20prefix)) instead, such that `getValue()` throws the `ParseException` directly?